cloud/blockstore/libs/endpoints/endpoint_manager.cpp

```diff
@@ -105,11 +106,27 @@ bool CompareRequests(
         && left.GetBurstTime() == right.GetBurstTime();
 }
 
+bool CompareNbdDevices(
+    const NProto::TStartEndpointRequest& lft,
+    const NProto::TStartEndpointRequest& rgt)
+{
+    if (lft.GetNbdDeviceFile() && rgt.GetNbdDeviceFile()) {
```

Comment:
doesn't empty string evaluates to bool true and this condition will always be true

Reply:
No, if one of NbdDeviceFiles will be empty then this condition will be false. There are tests for it.

Reply:
in this case what if we have `lft = ("", true) rgt=("/dev/nbd0", false)`
first if is false since `lft.GetNbdDeviceFile()` false
second if is false
adn we return `true`

Reply:
Yes, it's correct. User asked nbs to start endpoint with any free device, but nbs has already started the same endpoint with /dev/nbd0, so we return true here and user gets S_ALREADY in response.

Reply:
that's a little confusing since in CompareXX the two requests are different but the compare returns true ... maybe you should move this logic outside like here https://github.com/ydb-platform/nbs/blob/main/cloud/blockstore/libs/endpoints/endpoint_manager.cpp#L716
or at least put comment to explain the last comparison

Reply:
I've just refactored this place.